### ISSUE TITLE
Creating Route53 Hosted zone fails with InvalidClientTokenId

### ISSUE THREAD

**Issue by spstratis** [terraform-providers/terraform-provider-aws#8999]
Details below, but at a high level I've had no issues building out resources in AWS GovCloud, particularly in the US-WEST region. When I decided to add a resource for a private `aws_route53_zone` I get the below error ```* aws_route53_zone.private: error creating Route53 Hosted Zone: InvalidClientTokenId: The security token included in the request is invalid.
        status code: 403, request id: a9124a21-8eba-11e9-8bbb-c59c842ad843```

Normally I would think this is due to incorrect IAM creds since it's a 403, but my creds are working fine for every other resource, even those that are in the same TF file. I even tried changing them but no luck. Anyone know what could be the cause of this and how I can get around it. Route53 is supposed to be available in GovCloud us-west.

### Terraform Version

Terraform v0.11.13
+ provider.aws v2.12.0

### Affected Resource(s)

Route53 private hosted zone in us us-gov-west. 

### Terraform Configuration Files
provider "aws" {
  region     = "us-gov-west-1"
  access_key = "${var.aws_access_key}"
  secret_key = "${var.aws_secret_key}"
}

... Other VPC resources.

resource "aws_route53_zone" "private" {
  name = "my-domain.com"
  comment = "my-domain (preprod-gov) terraform"

  vpc = {
    vpc_id = "${module.preprod_gov_vpc.vpc_id}"
  }
}

### Debug Output

```
1 error(s) occurred:

* aws_route53_zone.private: 1 error(s) occurred:

* aws_route53_zone.private: error creating Route53 Hosted Zone: InvalidClientTokenId: The security token included in the request is invalid.
        status code: 403, request id: a9124a21-8eba-11e9-8bbb-c59c842ad843
```
### Expected Behavior

Successfully build the private zone. 

### Actual Behavior

Failing with the error mentioned above. 

### Steps to Reproduce

Added resource for aws_reoute53_zone and then terraform plan, terraform apply. 

### Important Factoids
Running in GovCloud but in US-West which should have R53 support

**Comment by bflad** [comment]
Hi @spstratis 👋 Sorry for the trouble here. Historically, Route 53 was only available in AWS Commercial, so the provider hardcoded the region for the service to `us-east-1` to simplify configurations for operators:

https://github.com/terraform-providers/terraform-provider-aws/blob/5daf292b7bfece60a41e65165a2cd77afa017fc6/aws/config.go#L428

Since Route 53 was [recently added to AWS GovCloud (US)](https://aws.amazon.com/about-aws/whats-new/2019/05/amazon-route-53-launches-private-dns-and-health-checks-in-the-aws-govcloud-us-west-region/), this assumption is no longer valid and this hardcoded region selection should only be performed when the Terraform AWS Provider is running in AWS Commercial.

A pull request has been submitted with the provider configuration fixes here: #9010

Unfortunately, using the Terraform AWS Provider [customizable endpoints](https://www.terraform.io/docs/providers/aws/guides/custom-service-endpoints.html) available in version 2.5.0 and later cannot serve as a workaround in this case because the previous provider initialization process forces the credentials region to `us-east-1`, which is not valid with the new `https://route53.us-gov.amazonaws.com` endpoint and will emit errors like the following when attempting API operations:

```
Error: error creating Route53 Hosted Zone: SignatureDoesNotMatch: Credential should be scoped to a valid region, not 'us-east-1'.
```

Hopefully the relatively small provider fix can get released fairly soon to handle this situation though.

**Comment by bflad** [comment]
The fix for this has been merged and will be released in version 2.16.0 of the Terraform AWS Provider, likely tomorrow. 👍

**Comment by bflad** [comment]
This has been released in [version 2.16.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2160-june-20-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
